src/components/rangeslider/draw.js

```diff
@@ -151,18 +157,29 @@ module.exports = function(gd) {
         // update current range
         setPixelRange(rangeSlider, gd, axisOpts, opts);
 
-        // update margins
+        // title goes next to range slider instead of tick labels, so
+        // just take it over and draw it from here
+        Titles.draw(gd, axisOpts._id + 'title', {
+            propContainer: axisOpts,
+            propName: axisOpts._name + '.title',
+            placeholder: fullLayout._dfltTitle.x,
+            attributes: {
+                x: axisOpts._offset + axisOpts._length / 2,
+                y: y + opts._height + opts._offsetShift + 10 + 1.5 * axisOpts.titlefont.size,
+                'text-anchor': 'middle'
+            }
+        });
```

Comment:
Where to put this `Titles.draw` call totally confused me. First I put it after `Plots.automargin` below... that failed, calling `Titles.draw` twice: first with the adjusted margins and THEN with the un-adjusted margins - the reason being that `Plots.autoMargin` invokes a whole new `Plotly.plot` call, and in that one we get the adjusted margin, but then *after* that we come back here and draw the title with the original margin. Instead, I had to put it here, so that the draw with the original margin happens before we adjust the margins, then the draw with the adjusted margin happens inside `Plots.autoMargin`. Or I suppose I could have added some MathJax so `Plots.autoMargin` would be async, that would have fixed it too 🙄 

@etpinard this is what got me thinking about altering our plotting pipeline with some intermediate step that's just concerned with arranging all the components but doesn't draw anything other than prerendering and stashing all the text we need sizes for.

Also, I started trying to pull out the `10 + 1.5 * axisOpts.titlefont.size` that's repeated from `Axes.doTicks` into some sort of `Titles.offset` function (that's why I started changing the structure in `titles/index`), but gave up for now as there's too much logic concerning side and other axis attributes in `doTicks`.